libmscore/measure.cpp

```diff
@@ -2676,13 +2682,24 @@ bool Measure::createEndBarLines()
                         ++aspan;
                         if (staff->show()) {          // update only if visible
                               bl->setSpan(aspan);
+                              // if the top staff was invisible
+                              // set spanFrom based on default for first visible staff
+                              if (show && spanFrom == unknownBarlineFrom)
```

Comment:
This is indeed the main change to address the issue as originally reported.  But to be clear: it is not about detecting a staff that was previously hidden and is now visible.  It's about the fact that at the moment when the spanFrom variable was set further up in the code, it was set based on the first staff of the span, and that staff is not visible.  That is, two staves, flute (5-line) and snare drum (1-line), the snapFrom was being set to 0 because that is appropriate for a flute staff, but that staff is not visible.  So when we get around to this line, actually setting the spanFrom for the barline, we were setting it to the appropriate value for the flute staff rather than the appropriate value for the snare drum staff.

In hindsight, we should probably just skip invisible staves completely when setting local span, spanFrom, and spanTo variables.  But as you say, this function is pretty convoluted, and I was reluctant to make any more significant changes to the flow.  I may take another look at this today after all.